synapse/rest/media/v1/media_storage.py

```diff
@@ -360,7 +359,7 @@ class ReadableFileWrapper:
     clock: Clock
     path: str
 
-    async def write_chunks_to(self, callback: Callable[[bytes], None]) -> None:
+    async def write_chunks_to(self, callback: Callable[[bytes], object]) -> None:
```

Comment:
mypy rejected passing `callback=BytesIO().write` here:

```
tests/rest/media/v1/test_media_storage.py:685: error: Argument 1 to "write_chunks_to" of "ReadableFileWrapper" has incompatible type "Callable[[bytes], int]"; expected "Callable[[bytes], None]"  [arg-type]
```

Callable is covariant in its return type: the `callback` we provide must return either `None` or a subtype of `None`.

But since we don't use the return value of `callback`, I've made this `object`. This works because every type is a subtype of object.

Reply:
I'd be OK with making it `int` if you want to be specific. That's what write is supposed to return.

Reply:
Possibly, I think I erred towards being generic here since we don't actually _use_ the return value of `callback` (currently)